app/src/main/kotlin/com/wire/android/ui/common/topappbar/AppTopBarWithSearchBarLayout.kt

```diff
@@ -0,0 +1,269 @@
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import com.wire.android.R
+import com.wire.android.ui.common.SearchBarInput
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.theme.wireColorScheme
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.scan
+
+
+/**
+ * AppTopBarWithSearchBarLayout is a "wrapper" around the [content] that should have a TopBar with a SearchBar
+ * on the Top of the [content]. To collapse the searchbar when the user scrolls down on the list within the [content]
+ * it is necessary to pass a [scrollPosition] usually it is a firstVisibleItemIndex coming out of the list within the content.
+ * AppTopBarWithSearchBarLayout also exposes [searchQuery] as well as [onSearchQueryChanged] so that we are able to manipulate
+ * the [searchQuery] outside of the AppTopBarWithSearchBarLayout for example a ViewModel. Beside collapsing the searchbar when scrolling
+ * through the list, AppTopBarWithSearchBarLayout also hides/shows the TopBar whenever the SearchBarInput is clicked so that we
+ * can transit to the state where we provide the search query to the searchbar.
+ */
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun AppTopBarWithSearchBarLayout(
+    scrollPosition: Int,
+    searchBarHint: String,
+    searchQuery: String,
+    onSearchQueryChanged: (String) -> Unit,
+    onSearchClicked: () -> Unit = {},
+    onCloseSearchClicked: () -> Unit = {},
+    content: @Composable () -> Unit,
+    appTopBar: @Composable () -> Unit,
+) {
+    ConstraintLayout(
+        Modifier
+            .fillMaxSize()
+    ) {
+        val (topBarRef, contentRef) = createRefs()
+
+        Box(
+            Modifier
+                .wrapContentSize()
+                .constrainAs(contentRef) {
+                    top.linkTo(topBarRef.bottom)
+                    bottom.linkTo(parent.bottom)
+
+                    height = Dimension.fillToConstraints
+                }
+        ) {
+            content()
+        }
+
+        Box(
+            Modifier
+                .wrapContentSize()
+                .constrainAs(topBarRef) {
+                    top.linkTo(parent.top)
+                    bottom.linkTo(contentRef.top)
+                }
+        ) {
+            AppTopBarWithSearchBar(
+                scrollPosition = scrollPosition,
+                searchBarHint = searchBarHint,
+                searchQuery = searchQuery,
+                onSearchQueryChanged = onSearchQueryChanged,
+                onSearchClicked = onSearchClicked,
+                onCloseSearchClicked = onCloseSearchClicked,
+                appTopBar = appTopBar
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun AppTopBarWithSearchBar(
```

Comment:
just think it is a bit cleaner this way AppTopBarWithSearchBarContent propagates all click events  and set the parameters and AppTopBarWithSearchBar manages those events also this way AppTopBarWithSearchBarContent is more resuable if needed because it only depends on some primities and callbacks